### PR TITLE
feat(argo-rollouts): Add rollout plugin gloo platform rbac

### PR DESCRIPTION
**Pull request by bgottfried91**
Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-rollouts/templates/controller/clusterrole.yaml [hunk 1]
```diff
@@ -267,5 +267,14 @@ rules:
...
     - update
 {{- end }}
+{{- if .Values.providerRBAC.providers.glooPlatform }}
+  # Access needed when using the Gloo Platform provider
+- apiGroups:
+    - networking.gloo.solo.io
+  resources:
+    - routetables
+  verbs:
+    - '*'
```
Comment (yu-croco) [inline charts/argo-rollouts/templates/controller/clusterrole.yaml:277]: Hi @bgottfried91 , can you please give us the doc what you refer for this config?

Comment (pdrastil) [reply charts/argo-rollouts/templates/controller/clusterrole.yaml:277]: Hi @yu-croco - this resource is used for traffic delegation via RouteTables for Gloo Edge
Ref: https://docs.solo.io/gloo-edge/latest/introduction/traffic_management/

Comment (pdrastil) [reply charts/argo-rollouts/templates/controller/clusterrole.yaml:277]: The plugin is mentioned in values.yaml 
Ref: https://github.com/argoproj-labs/rollouts-plugin-trafficrouter-glooplatform/blob/main/README.md

Comment (yu-croco) [reply charts/argo-rollouts/templates/controller/clusterrole.yaml:277]: Hi @pdrastil , thank you for the info ! 👍

Comment (pdrastil) [reply charts/argo-rollouts/templates/controller/clusterrole.yaml:277]: @yu-croco FYI - it seems that Gloo is adopting new Kubernetes Gateway API
Ref: https://github.com/solo-io/gloo/tree/v2.0.x

**Comment by yu-croco** [comment]
Hi @bgottfried91 , please fix failed CI (I think [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation) will help you), otherwise LGTM. 👍

File: charts/argo-rollouts/templates/controller/clusterrole.yaml (hunk 1 above)
Comment (bgottfried91) [reply charts/argo-rollouts/templates/controller/clusterrole.yaml:277]: Just as a FYI, this addition is to support the plugin for Gloo Platform. There's a separate plugin for Gloo Edge, they're different APIs. We'll probably submit an update to support adding the RBAC for that plugin soon!

This is the overview page for Gloo Platform's traffic management (https://docs.solo.io/gloo-mesh-enterprise/latest/concepts/traffic-management/) and the API reference for it (https://docs.solo.io/gloo-mesh-enterprise/latest/reference/api/route_table/)

Petr's correct that GE builds on the Gateway API now though!

Comment (yu-croco) [reply charts/argo-rollouts/templates/controller/clusterrole.yaml:277]: Thank you for the info :)

**Review by yu-croco — APPROVED** [review]
Thank you for your contribution. LGTM 🚀

**Review by jmeridth — APPROVED** [review]

**Pull request merged**